ltr/concourse/fetch.sh

```diff
@@ -11,6 +11,18 @@ S3_BUCKET="govuk-${GOVUK_ENVIRONMENT}-search-relevancy"
 GIT_BRANCH="deployed-to-${GOVUK_ENVIRONMENT}"
 AWS_REGION="eu-west-1"
 
+function scale_down_asg {
+  # any previous assumption may have timed out, so assume again before scaling down the ASG
+  assume_role
+
+  echo "Scaling down ASG..."
+  aws autoscaling set-desired-capacity \
+      --region "$AWS_REGION" \
+      --auto-scaling-group-name "$EC2_NAME" \
+      --desired-capacity 0
+}
+trap scale_down_asg EXIT
```

Comment:
It might be worth moving the cleanup step into a separate [`ensure` step](https://concourse-ci.org/jobs.html#schema.step.ensure) in the Concourse fetch job.

The benefit would be that we'd still capture the status of the job (so the train job would get triggered correctly depending on the exit code of the fetch job), and the turn-down of the asg would always run.

```yaml
plan:
- task: fetch
  file: fetch-job.yml
  ensure:
    task: turn-down-fetch-machine
    file: turn-down.yml
```

Reply:
ooh, nice idea 👍

Reply:
@bilbof would you mind having another look at this please?